streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -102,6 +102,7 @@ public void before() {
         props = new Properties();
         props.put(StreamsConfig.APPLICATION_ID_CONFIG, "appId");
         props.put(StreamsConfig.CLIENT_ID_CONFIG, "clientId");
+        props.put(StreamsConfig.APPLICATION_SERVER_CONFIG, "localhost:" + String.valueOf(Math.round(Math.random() * 4460)));
```

Comment:
Why do we need this?

Reply:
Because as suggested by @guozhangwang we don't to change Streams Metadata, so I am recreating partitionsByHost from tasksByHost in onAssignment(). While doing that, if we don't pass any application server config, the hostinfo is null, all the hosts overwrite on each other in the partitionsByHost map and most of the test cases fail.

Reply:
Not sure if I understand. If no `application.server` is given, `partitionsByHost` should be empty even before your change -- so why does it break? Should `partitionsByHost ` not be empty after this change, too?